### ISSUE TITLE
Raspbian MagicMirror Node.js Problem

### ISSUE THREAD

**Issue by Stefan9684** [sdetweil/MagicMirror_scripts#101]
Guten Morgen,

leider verzweifele  ich langsam. Ich stoße immer wieder auf diese Problem. Habe schon verschiedene Versionen von Node installiert und wieder deinstalliert, finde aber keine Lösung. Bitte um Hilfe.....

install log being saved to /home/stefan/install.log
Updating packages ...
W: http://raspbian.raspberrypi.com/raspbian/dists/bookworm/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
[       <=>                                                                                                                                                                        ]
Installing helper tools ...
W: http://raspbian.raspberrypi.com/raspbian/dists/bookworm/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
bash: line 273: /usr/local/bin/node: cannot execute: required file not found
node failed to execute, please check the error message and resolve before trying again
stefan@raspberrypi:~ $

![Installation MM](https://github.com/user-attachments/assets/077a8283-28b2-4625-b537-d9f160431b89)

**Comment by sdetweil** [comment]
can you tell me what OS image this is?   I know it says bookworm, but bookworm is the current release

run commands
uname -a
lsb_release -a

show results

**Comment by Stefan9684** [comment]
![image](https://github.com/user-attachments/assets/9a5c5b62-d470-4e2c-999c-474bf9566d1b)

**Comment by Stefan9684** [comment]
Ok. So i have to wait?

**Comment by sdetweil** [comment]
Yes, or do the install manually. I have not seen this before, so will have to work on it. Have a couple things to do yet, before I can

**Comment by Stefan9684** [comment]
I find a Solution.
Install RasbianOS complete new. Install Older Version of Node.
Then Start your Script. Now it works

**Comment by sdetweil** [comment]
Can uou attach the install.log here

Thanks

**Comment by Stefan9684** [comment]
[installMM.txt](https://github.com/user-attachments/files/17982222/installMM.txt)

**Comment by sdetweil** [comment]
Thx
Got the same warning
![IMG_0509](https://github.com/user-attachments/assets/47ffc7d0-b7ca-4a21-9aaf-f59fb83d2f34)

**Comment by sdetweil** [comment]
ok, this is the key 
| Install RasbianOS complete new

the doc used to say complete... hmm

**Comment by Lateadapter** [comment]
Many thanks for the amazing work done here!

I can confirm, that the recently modified script (fix apt update fail if repo list corrupt, move to node 20.18.1, for release 2.30.0) 
did solve the issue using the latest Raspian.OS "2024-11-19-raspios-bookworm-arm64.img".

Nevertheless i would suggest a small hint in readme, that a full OS with Desktop is required as precondition. I did start with the "Lite"-version and failed...
Even looking with Google for "MagicMirror installation" will end up with following landing page and the very important hint in "Requirement" section will be easily missed:
[https://docs.magicmirror.builders/getting-started/installation.html](url)

A second point - it would be helpfull to have in the script the option for "serveronly" or "clientonly" mode. I actual was looking for a clientonly installation .... and finally succeded also after some additional trial and error. Regarding the background - I'm using MagicMirror already few years as docker server on Synology-NAS and a Raspi with FullPageOS as client - untouched very long time. However recently probably after some docker update the client became instable. First idea was a damaged SD-card but a reinstallation with latest kiosk made the situation even worse - meanwhile done in few iterations. 
I now suspect, that the latest server versions of MagicMirror electron at one side and chromium (or the implemented version in FullPageOS) at other side have some interaction issues. Or probably I'm just missing the knowledge to find the real issue.
Therfore my idea was, to use at client side the exact same electron version as at server side .... 

And it looks very promising so far!

**Comment by sdetweil** [comment]
see requirements in the MagicMirror doc

![IMG_0602](https://github.com/user-attachments/assets/6a1f2cd5-19e6-47a9-972d-0d9a8c04923d)

many of the latest updates and 3rd party modules  use the latest js language idioms. they require an updated electron or chromium/firefox/... browser

you no longer meed the MagicMirror client. just start a compatible browser
